common/persistence/cassandraVisibilityPersistence.go

```diff
@@ -150,7 +152,7 @@ func (v *cassandraVisibilityPersistence) Close() {
 
 func (v *cassandraVisibilityPersistence) RecordWorkflowExecutionStarted(
 	request *RecordWorkflowExecutionStartedRequest) error {
-	ttl := request.WorkflowTimeout + openExecutionTTLBuffer
+	ttl := common.MinInt64(request.WorkflowTimeout+openExecutionTTLBuffer, maxCassandraTTL)
```

Comment:
Open visibility record will be deleted after this TTL.  It will be very confusing if we allow WorkflowTimeout of greater than 20 years but only allow visibility record for upto 20 years max.  I think we should just drop the TTL part from the query if Workflow Timeout is bigger than the supported TTL.